invenio_rdm_records/views.py

```diff
@@ -33,6 +49,80 @@ def init(state):
     iregistry.register(ext.records_service.draft_indexer, indexer_id="records-drafts")
 
 
+@blueprint.route("/access-requests/verify")
+def verify_access_request_token():
+    """UI endpoint for verifying guest access request tokens.
+
+    When the token is verified successfully, a new guest access request will be created
+    and the token object will be deleted from the database.
+    The token value will be stored with the newly created request and grant access
+    permissions to the request details.
+    """
+    # TODO check how the email verification logic works and get inspiration from there
+    token = request.args.get("access_request_token")
+    try:
+        access_request = current_service.access.create_guest_access_request(
+            identity=g.identity, token=token
+        )
+    except DuplicateAccessRequestError:
+        # TODO should we redirect to the already existing request? would that leak data?
+        # TODO we should still delete the guest access request token
+        abort(409)
```

Comment:
Comment: I think we agreed on redirecting, there should not be any data leaking as the token is send via email.

Reply:
True, I knew I missed something :sweat_smile:

Reply:
Fixed!